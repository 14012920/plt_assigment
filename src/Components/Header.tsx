import React from "react";
import { View, Pressable } from "react-native";
import { Appbar, Text, withTheme } from "react-native-paper";
import Ionicons from "@expo/vector-icons/Ionicons";

type hederProps = {
  title: string;
  cartCount?: number;
  onpressCart?: () => void;
  theme: any;
  type: string;
  goBack?: () => void;
  cartTotal?: number;
};

const Header = ({
  title,
  cartCount,
  onpressCart,
  theme,
  goBack,
  type,
  cartTotal,
}: hederProps) => {
  if (type === "products") {
    return (
      <Appbar.Header>
        <Appbar.Content title={title} />
        <View>
          <Pressable onPress={onpressCart}>
            <Ionicons
              Ionicons
              name="md-cart"
              size={32}
              color={theme.colors.iconColor}
            />
            {cartCount ? (
              <View
                style={{
                  position: "absolute",
                  backgroundColor: "#192345",
                  width: 16,
                  height: 16,
                  borderRadius: 15 / 2,
                  right: 1,
                  top: +4,
                  alignItems: "center",
                  justifyContent: "center",
                }}
              >
                <Text
                  style={{
                    alignItems: "center",
                    justifyContent: "center",
                    color: "#FFFFFF",
                    fontSize: 8,
                  }}
                >
                  {cartCount}
                </Text>
              </View>
            ) : null}
          </Pressable>
        </View>
      </Appbar.Header>
    );
  } else {
    return (
      <Appbar.Header>
        <Appbar.BackAction onPress={goBack} />
        <Appbar.Content title={title} />
        <View style={{ padding: 4 }}>
          <Text variant="labelLarge">Total:â‚¹{cartTotal}</Text>
        </View>
      </Appbar.Header>
    );
  }
};
export default withTheme(Header);
