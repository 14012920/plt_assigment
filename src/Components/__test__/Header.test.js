import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import Header from "../Header";

jest.mock("react-native-safe-area-context", () => {
  const inset = { top: 0, right: 0, bottom: 0, left: 0 };

  return {
    SafeAreaProvider: jest.fn().mockImplementation(({ children }) => children),
    SafeAreaConsumer: jest
      .fn()
      .mockImplementation(({ children }) => children(inset)),
    useSafeAreaInsets: jest.fn().mockImplementation(() => inset),
  };
});

describe("render Header Screen", () => {
  beforeEach(() => {});

  test("render Header Screen", () => {
    const props = {
      title: "Products",
      cartCount: 0,
      onpressCart: jest.fn(),
      goBack: jest.fn(),
      type: "products",
      cartTotal: 0,
    };
    const tree = renderer.create(<Header {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("render Header for product Screen", () => {
    const props = {
      title: "Products",
      cartCount: 0,
      onpressCart: jest.fn(),
      goBack: jest.fn(),
      type: "products",
      cartTotal: 0,
    };
    const { getByText } = render(<Header {...props} />);
    expect(getByText("Products")).toBeTruthy();
  });
});
