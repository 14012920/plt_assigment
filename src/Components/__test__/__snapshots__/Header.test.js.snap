// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header Screen 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(255, 251, 254, 1)",
        "elevation": 0,
        "paddingHorizontal": 0,
        "paddingTop": 0,
        "zIndex": 0,
      },
      {},
    ]
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "flexDirection": "row",
            "height": 64,
            "paddingBottom": undefined,
            "paddingHorizontal": 4,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
          }
        }
      >
        <View
          style={
            {
              "width": 52,
            }
          }
        />
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          pointerEvents="box-none"
          style={
            [
              {
                "flex": 1,
                "paddingHorizontal": 12,
              },
              {
                "paddingHorizontal": 0,
              },
              [
                false,
                {
                  "alignItems": "center",
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            accessibilityRole="header"
            accessible={true}
            numberOfLines={1}
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                [
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 28,
                  },
                  false,
                  undefined,
                ],
              ]
            }
          >
            Products
          </Text>
        </View>
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`render Header Screen render Header Screen 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(255, 251, 254, 1)",
        "elevation": 0,
        "paddingHorizontal": 0,
        "paddingTop": 0,
        "zIndex": 0,
      },
      {},
    ]
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "flexDirection": "row",
            "height": 64,
            "paddingBottom": undefined,
            "paddingHorizontal": 4,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
          }
        }
      >
        <View
          style={
            {
              "width": 52,
            }
          }
        />
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          pointerEvents="box-none"
          style={
            [
              {
                "flex": 1,
                "paddingHorizontal": 12,
              },
              {
                "paddingHorizontal": 0,
              },
              [
                false,
                {
                  "alignItems": "center",
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            accessibilityRole="header"
            accessible={true}
            numberOfLines={1}
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                [
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 28,
                  },
                  false,
                  undefined,
                ],
              ]
            }
          >
            Products
          </Text>
        </View>
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
